src/components/SearchableMultiSelect/SearchableMultiSelect.jsx

```diff
@@ -357,26 +399,54 @@ const SearchableMultiSelect = createClass({
 			flattenedOptionsData,
 		} = this.state;
 
-		const isAllOptionsHidden = _.every(
+		const visibleOptions = _.reject(
 			flattenedOptionsData,
-			({ optionProps }) => !optionFilter(searchText, optionProps)
+			'optionProps.isHidden'
 		);
 
+		const isAllOptionsHidden = _.isEmpty(visibleOptions);
+
+		const isEveryVisibleOptionSelected = _.every(visibleOptions, ({
+			optionIndex,
+		}) => _.includes(selectedIndices, optionIndex));
+
+		const isAnyVisibleOptionSelected = _.some(visibleOptions, ({
+			optionIndex,
+		}) => _.includes(selectedIndices, optionIndex));
```

Comment:
there's some relatively low-hanging optimization fruit here, i.e. pulling these four mappings into a single, much less legible reduce. Might be worth it, since this is in the rendering hot path.